src/pixel.rs

```diff
@@ -153,33 +158,21 @@ impl Pixels {
             Pixels::Indexed(v) => v.len(),
         }
     }
-}
-
-pub(crate) fn resolve_indexed(
-    pixel: &Indexed,
-    palette: &ColorPalette,
-    transparent_color_index: u8,
-    layer_is_background: bool,
-) -> Result<Rgba> {
-    pixel
-        .as_rgba(palette, transparent_color_index, layer_is_background)
-        .ok_or_else(|| {
-            AsepriteParseError::InvalidInput(format!(
-                "Index out of range: {} (max: {})",
-                pixel.value(),
-                palette.num_colors()
-            ))
-        })
-}
 
-pub(crate) fn resolve_indexed_pixels(
-    pixels: &[Indexed],
-    palette: &ColorPalette,
-    transparent_color_index: u8,
-    layer_is_background: bool,
-) -> Result<Vec<Rgba>> {
-    pixels
-        .iter()
-        .map(|px| resolve_indexed(px, palette, transparent_color_index, layer_is_background))
-        .collect()
+    pub(crate) fn clone_as_image_rgba<F>(&self, index_resolver: F) -> Vec<image::Rgba<u8>>
```

Comment:
Hm, I wonder why we need our own `Rgba` type. I think we could just reuse/re-export the one from `image` since that will remain a required dependency. Then in this function you could return a `std::borrow::Cow` so that we don`t need to copy in case the image data is already in rgba.